packages/vitest/src/runtime/loader.ts

```diff
@@ -18,15 +27,31 @@ export const resolve: Resolver = async (url, context, next) => {
   if (resolver) {
     const resolved = await resolver(id, importer)
     if (resolved) {
-      return {
-        url: pathToFileURL(resolved.id).toString(),
+      const filepath = pathToFileURL(resolved.id).toString()
+      const result: ResolveResult = {
+        url: filepath,
         shortCircuit: true,
       }
+      const source = cache.get(resolved.id)?.source ?? await readFile(resolved.id, 'utf8')
```

Comment:
Maybe we could cache in disk based on package hash (we might be able read it from Vite)

Reply:
But why, they are already on disk 👀 Are you afraid it will not update, if I change something inside? If so, it won't update either way, because HMR doesn't work for node_modules 👀

Reply:
Oh I didn't notice we need the content 😅